File: fixtures/hub-gold-rails-routes/app/controllers/hub_controller.rb
# frozen_string_literal: true
# Thin ActionController peels for hub-gold-rails-routes (G10115 / D6540).
# Only render json: + params[:id] — no filters / views / DI invent (D6447).

class HubController < ActionController::API
  def health
    render json: true
  end

  def ping
    render json: 42
  end

  def version
    render json: 1
  end

  def ready
    render json: "ok"
  end

  def count
    render json: 3
  end

  def flag
    render json: "chrysalis"
  end

  def build
    render json: 2026
  end

  def tier
    render json: "gold"
  end

  def meta
    render json: { service: "hub-gold-rails-routes", version: 1 }
  end

  def echo
    render json: { echo: true }
  end

  def search
    render json: { q: params[:q] || "" }
  end

  def user
    render json: params[:userId]
  end

  def stats
    render json: 3
  end

  def notify
    render json: { ok: true }, status: 202
  end
end

File: fixtures/hub-gold-phoenix-controllers/lib/hub_gold_web/controllers/hub_controller.ex
defmodule HubGoldWeb.HubController do
  use Phoenix.Controller

  # Thin Phoenix.Controller peels for hub-gold-phoenix-controllers (G10126 / D6540).
  # Only json/2 + put_status/2 + params["…"] — no LiveView / views / plugs invent (D6447).

  def health(conn, _params) do
    json(conn, true)
  end

  def ping(conn, _params) do
    json(conn, 42)
  end

  def version(conn, _params) do
    json(conn, 1)
  end

  def ready(conn, _params) do
    json(conn, "ok")
  end

  def count(conn, _params) do
    json(conn, 3)
  end

  def flag(conn, _params) do
    json(conn, "chrysalis")
  end

  def build(conn, _params) do
    json(conn, 2026)
  end

  def tier(conn, _params) do
    json(conn, "gold")
  end

  def meta(conn, _params) do
    json(conn, %{service: "hub-gold-phoenix-controllers", version: 1})
  end

  def echo(conn, _params) do
    json(conn, %{echo: true})
  end

  def search(conn, params) do
    json(conn, %{q: params["q"] || ""})
  end

  def user(conn, params) do
    json(conn, params["userId"])
  end

  def stats(conn, _params) do
    json(conn, 3)
  end

  def notify(conn, _params) do
    conn
    |> put_status(202)
    |> json(%{ok: true})
  end
end
